Common/cpp/Registries/EventHandlerRegistry.cpp

```diff
@@ -1,14 +1,23 @@
 #include "EventHandlerRegistry.h"
 #include "WorkletEventHandler.h"
 
+#include <utility>
+
 namespace reanimated {
 
 void EventHandlerRegistry::registerEventHandler(
     std::shared_ptr<WorkletEventHandler> eventHandler) {
   const std::lock_guard<std::mutex> lock(instanceMutex);
   const auto &eventName = eventHandler->getEventName();
+  const auto emitterReactTag = eventHandler->getEmitterReactTag();
   auto handlerId = eventHandler->getHandlerId();
-  eventMappings[eventName][handlerId] = eventHandler;
+
+  if (emitterReactTag == -1) {
```

Comment:
I think that i would be better if we replace condition `emitterReactTag == -1` with something like: `eventHandler->ignoreEmitterReactTag()` / `eventHandler->useOnlyEventName()` etc.